src/System.Collections.Immutable/src/System/Collections/Immutable/ImmutableExtensions.cs

```diff
@@ -85,6 +85,38 @@ internal static int GetCount<T>(ref IEnumerable<T> sequence)
 
             return count;
         }
+        
+        /// <summary>
+        /// Copies part of an IEnumerable&lt;T&gt; to an array, starting at the specified index.
+        /// </summary>
+        /// <typeparam name="T">The type of element.</typeparam>
+        /// <param name="sequence">The sequence of elements you want to copy.</param>
+        /// <param name="dest">The destination array.</param>
+        /// <param name="index">The index in the array to start copying to.</param>
+        internal static void CopyTo(this IEnumerable<T> sequence, T[] dest, int index)
+        {
+            var collectionOfT = items as ICollection<T>;
+            
+            if (collectionOfT != null)
+            {
+                collectionOfT.CopyTo(tmp, index);
+                return;
+            }
+            
+            var collection = items as ICollection;
+            
```

Comment:
style: IMO it's better to omit this blank line between the conditional cast and the if that will use the result. It makes reading the method more obvious that you have three chunks of work.